drivers/can/can_stm32h7_fdcan.c

```diff
@@ -94,22 +109,32 @@ static int can_stm32h7_clock_enable(const struct device *dev)
 	const struct device *const clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
 	int ret;
 
-	LL_RCC_SetFDCANClockSource(LL_RCC_FDCAN_CLKSOURCE_PLL1Q);
-
 	if (!device_is_ready(clk)) {
 		LOG_ERR("clock control device not ready");
 		return -ENODEV;
 	}
 
-	ret = clock_control_on(clk, (clock_control_subsys_t)&stm32h7_cfg->pclken);
+	if (IS_ENABLED(STM32H7_FDCAN_DOMAIN_CLOCK_SUPPORT) && (stm32h7_cfg->pclk_len > 1)) {
+		ret = clock_control_configure(clk,
+				(clock_control_subsys_t)&stm32h7_cfg->pclken[1],
+				NULL);
+		if (ret < 0) {
+			LOG_ERR("Could not select can_stm32fd domain clock");
+			return ret;
+		}
+	}
+
+	ret = clock_control_on(clk, (clock_control_subsys_t)&stm32h7_cfg->pclken[0]);
 	if (ret != 0) {
 		LOG_ERR("failure enabling clock");
 		return ret;
 	}
 
-	if (!LL_RCC_PLL1Q_IsEnabled()) {
-		LOG_ERR("PLL1Q clock must be enabled!");
-		return -EIO;
+	if (stm32h7_cfg->clock_divider != 0U) {
```

Comment:
As I have it in my memory - you are allowed write 0 in mentioned register, cause the HW will add +1 anyway. Could you check it in RM please?

Reply:
The check for `0` is to allow for not configuring the divider (as this is shared between all instances). Writing `0` would mean a divisor of `1`, hence the bitshifting.